File: Level-0/58.java
class Solution {
    public int solution(int[][] board) {
        int answer = 0;
        int[] dx = {1, 1, 1, 0, 0, -1, -1, -1};
        int[] dy = {-1, 0, 1, -1, 1, -1, 0, 1};
        for(int x = 0; x < board.length; x++){
            for(int y = 0; y < board[x].length; y++){
                if(board[x][y] == 1){
                    for(int i = 0; i < 8; i++){
                        int nx = x + dx[i];
                        int ny = y + dy[i];
                        if(nx < 0 || ny < 0 || nx >= board.length || ny >= board.length) continue;
                        if(board[nx][ny] != 0) continue;
                        board[nx][ny] = board[x][y] + 1;
                    }
                }
            }
        }
        
        for(int i = 0; i < board.length; i++){
            for(int j = 0; j < board[i].length; j++){
                if(board[i][j] == 0) answer++;
            }
        }
        return answer;
    }
}

File: Level-0/58.js
function solution(board) {
    const n = board.length;
    
    const dx = [-1, -1, -1, 0, 0, 1, 1, 1];
    const dy = [-1, 0, 1, -1, 1, -1, 0, 1];
    
    let answer = n * n;
    
    for(let r = 0; r < n; r++) {
        for(let c = 0; c < n; c++) {
            if(board[r][c] === 0) {
                for(let d = 0; d < 8; d++) {
                    let nr = r + dx[d];
                    let nc = c + dy[d];
                    
                    if(nr < 0 || nr >= n || nc < 0 || nc >= n) continue;
                    if(board[nr][nc] === 1) {
                        answer--;
                        break;
                    }
                }
            }else {
                answer--;
                continue;
            }
        }
    }
    
    return answer;
}
